app/views/users/show.html.haml

```diff
@@ -19,9 +19,19 @@
             %strong Cost Description
             -@cost_description.each do |cost,item|
               %p="-- #{cost} #{item}"
-            ="Total Cost: $#{@application_cost}"
+            ="Total Cost: #{number_to_currency(@application_cost)}"
+        %p
+        %strong= "Payment Status: "
+        -if @application.hasPaid? and @application.pay_status == 2
+          ="Payment received. Thank you!"
+        -elsif @application.pay_status == 3
```

Comment:
I think the constants defined in Application would be helpful here and take this logic into the controller instead. An example of taking the logic to the controller is how Application submission status is done.